### DISCUSSION TITLE
Create Tabular Data Package with CLI

### DISCUSSION THREAD

**Discussion by rgaiacs** [frictionlessdata/frictionlessdata.io#746] [Q&A]
I can use `frictionless describe` to generate a `datapackage.json` for me, for example

```
$ frictionless describe --type package --json iris.csv
{
  "resources": [
    {
      "path": "iris.csv",
      "name": "iris",
      "profile": "tabular-data-resource",
      "scheme": "file",
      "format": "csv",
      "hashing": "md5",
      "encoding": "utf-8",
      "schema": {
        "fields": [
          {
            "type": "number",
            "name": "Sepal.Length"
          },
          {
            "type": "number",
            "name": "Sepal.Width"
          },
          {
            "type": "number",
            "name": "Petal.Length"
          },
          {
            "type": "number",
            "name": "Petal.Width"
          },
          {
            "type": "string",
            "name": "Species"
          }
        ]
      }
    }
  ],
  "profile": "data-package"
}
```

How do I change `"profile": "data-package"` to `"profile": "tabular-data-package"`?

**Comment by aborruso** [comment]
> How do I change `"profile": "data-package"` to `"profile": "tabular-data-package"`?

Does it exist? We do not have it here https://framework.frictionlessdata.io/docs/guides/describing-data/#describe-functions

**Comment by rgaiacs** [comment]
Hi @aborruso.

> Does it exist?

https://specs.frictionlessdata.io/ has [Data Package](https://specs.frictionlessdata.io/data-package/) and [Tabular Data Package](https://specs.frictionlessdata.io/tabular-data-package/). Tabular Data Package specs says

> There MUST be a profile property with the value tabular-data-package

So, I was expecting to to be possible to set the `profile to `tabular-data-package`.

Is this related to v1 vs v1.1 or v2 of Frictionless Data?

**Comment by roll** [comment]
Hi @rgaiacs, the framework doesn't add the `tabular-data-package` profile as it might be too restrcting and we already have resource marked as `tabular-data-resource`. TBH the concept of Tabular Data Package after intorucing Tabular Data Resources is not fully clear to me

**Comment by rgaiacs** [comment]
> the framework doesn't add the tabular-data-package profile as it might be too restrcting and we already have resource marked as tabular-data-resource.

I understand this part.

> TBH the concept of Tabular Data Package after intorucing Tabular Data Resources is not fully clear to me

I'm studying the ["reference" implementation in R](https://github.com/frictionlessdata/frictionless-r) and [it has](https://github.com/frictionlessdata/frictionless-r/blob/c0869654c1bae1d25782c0315fa81f96fa9fa291/R/create_package.R) `profile = "tabular-data-package"`.

Based on your comment, my understanding is that the R implementation could use `"profile": "data-package"`.

**Reply by roll** [reply]
Yes it could. Tabular Data Package requires all the resources to be a tabular CSV which is quite a strong condition. In real-life majority of datasets has tables AND supporting files like PDFs/etc so in my opinion after introducing Tabular Data Resource concept using Tabular Data Package is not really needed anymore
